contracts/Zauction.sol

```diff
@@ -12,11 +12,15 @@ contract ZAuction is Initializable, OwnableUpgradeable {
   using ECDSA for bytes32;
 
   IERC20 public token;
+  IERC721 public nft;
```

Comment:
The `registrar` will always be the same as `nft`
You can update `IRegistrar` to extend/implement `IERC721`.